File: IntelliJ IDEA/Java/src/demo/itemsControlImplementCustomViewList/ViewModel.java
package demo.itemsControlImplementCustomViewList;

import com.eruru.mvvm.MVVMObservableCollection;
import com.eruru.mvvm.MVVMRelayCommand;

public class ViewModel {

	private int id;
	private Item tempItem;
	private MVVMObservableCollection<Item> items = new MVVMObservableCollection<Item> () {{

		add (new Item (0));
		add (new Item (0));
		add (new Item (0));

	}};
	private MVVMRelayCommand onAdd = new MVVMRelayCommand (value -> {
		items.add (newItem ());
	});
	private MVVMRelayCommand onSetNew = new MVVMRelayCommand (value -> {
		items.set (items.indexOf ((Item) value), newItem ());
	});
	private MVVMRelayCommand onMove = new MVVMRelayCommand (value -> {
		Item item = (Item) value;
		if (tempItem == null) {
			tempItem = item;
			return;
		}
		int oldIndex = items.indexOf (tempItem);
		tempItem = null;
		if (oldIndex < 0) {
			return;
		}
		items.move (oldIndex, items.indexOf (item));
	});
	private MVVMRelayCommand onDelete = new MVVMRelayCommand (value -> {
		items.remove ((Item) value);
	});
	private MVVMRelayCommand onInsert = new MVVMRelayCommand (value -> {
		Item item = (Item) value;
		items.add (items.indexOf (item), newItem ());
	});
	private MVVMRelayCommand onClear = new MVVMRelayCommand (value -> {
		items.clear ();
	});

	public MVVMObservableCollection<Item> getItems () {
		return items;
	}

	public void setItems (MVVMObservableCollection<Item> items) {
		this.items = items;
	}

	public MVVMRelayCommand getOnAdd () {
		return onAdd;
	}

	public void setOnAdd (MVVMRelayCommand onAdd) {
		this.onAdd = onAdd;
	}

	public MVVMRelayCommand getOnSetNew () {
		return onSetNew;
	}

	public void setOnSetNew (MVVMRelayCommand onSetNew) {
		this.onSetNew = onSetNew;
	}

	public MVVMRelayCommand getOnMove () {
		return onMove;
	}

	public void setOnMove (MVVMRelayCommand onMove) {
		this.onMove = onMove;
	}

	public MVVMRelayCommand getOnDelete () {
		return onDelete;
	}

	public void setOnDelete (MVVMRelayCommand onDelete) {
		this.onDelete = onDelete;
	}

	public MVVMRelayCommand getOnClear () {
		return onClear;
	}

	public void setOnClear (MVVMRelayCommand onClear) {
		this.onClear = onClear;
	}

	private Item newItem () {
		return new Item (id++);
	}

	public MVVMRelayCommand getOnInsert () {
		return onInsert;
	}

	public void setOnInsert (MVVMRelayCommand onInsert) {
		this.onInsert = onInsert;
	}

}

File: Unity/Assets/Eruru.MVVM/Demos/ItemsControl/ViewModel.cs
using UnityEngine;
using System.Collections;

namespace Eruru.MVVM.Demo.ItemsControl {

	public class ViewModel {

		public MVVMObservableCollection<Item> Items { get; set; }
		public MVVMRelayCommand OnAdd { get; set; }
		public MVVMRelayCommand OnEdit { get; set; }
		public MVVMRelayCommand OnDelete { get; set; }

		public ViewModel () {
			Items = new MVVMObservableCollection<Item> () {
				new Item ("小明", 18, "光明小学", "三好学生"),
				new Item ("小刚", 20, "复兴中学", "优秀班干部"),
				new Item ("吉姆格林", 19, "光明小学", "吉姆做了汽车公司经理"),
				new Item ("李雷", 25, "复兴中学", "不老实的家伙"),
				new Item ("韩梅梅", 22, "光明小学", "在一起")
			};
			OnAdd = new MVVMRelayCommand (value => {
				AddView addView = Object.Instantiate (Resources.Load<AddView> ("ItemsControl/WindowAdd"));
				addView.Item = new Item ();
				MVVMControl control = addView.Build ();
				addView.OnConfirm = () => {
					Items.Add (control.DataContext.GetTargetValue<AddViewModel> ().Item);
				};
			});
			OnEdit = new MVVMRelayCommand (value => {
				AddView addView = Object.Instantiate (Resources.Load<AddView> ("ItemsControl/WindowAdd"));
				Item item = (Item)value;
				addView.Item = item.Clone ();
				MVVMControl control = addView.Build ();
				addView.OnConfirm = () => {
					Items[Items.IndexOf (item)] = control.DataContext.GetTargetValue<AddViewModel> ().Item;
				};
			});
			OnDelete = new MVVMRelayCommand (value => {
				Item item = (Item)value;
				Items.Remove (item);
			});
		}

	}

}